public/app/features/trails/Integrations/interpolation.ts

```diff
@@ -0,0 +1,56 @@
+import { AdHocVariableFilter, DataSourceInstanceSettings } from '@grafana/data';
+import { PromQuery, PrometheusDatasource } from '@grafana/prometheus';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { AdHocFiltersVariable, SceneVariable } from '@grafana/scenes';
+import { DataQuery, DataSourceRef } from '@grafana/schema';
+import { DashboardModel } from 'app/features/dashboard/state';
+import { DashboardScene } from 'app/features/dashboard-scene/scene/DashboardScene';
+
+export function interpolateVariables(
```

Comment:
you can replace all this implementation ds.interpolateVariablesInQueries  (this is a function on the datasource instance). 

This what we do when we go from dashboard to Explore or from data trails to Explore

https://github.com/grafana/grafana/blob/main/public/app/core/utils/explore.ts#L81